sympy/relation/eqntools/simplify.py

```diff
@@ -0,0 +1,140 @@
+"""
+Module for equation simplification.
+"""
+
+from functools import singledispatch
+
+from sympy.core import Add, Expr, S
+from sympy.core.compatibility import ordered
+from sympy.polys import Poly, poly, PolynomialError, gcd
+from sympy.solvers.solveset import linear_coeffs
+
+from sympy.relation.binrel import BinaryRelation
+from sympy.relation.equality import Equal
+
+
+@singledispatch
+def eqnsimp(rel, lhs, rhs, **kwargs):
+    raise NotImplementedError("Unknown binary predicate.")
+
+
+@eqnsimp.register(BinaryRelation)
+def _(rel, lhs, rhs, **kwargs):
+    eqn = rel(lhs, rhs)
+
+    r = rel(lhs.simplify(**kwargs), rhs.simplify(**kwargs))
+    if not isinstance(r.lhs, Expr) or not isinstance(r.rhs, Expr):
+        return r
+    dif = r.lhs - r.rhs
+    r = r.canonical
+    # If there is only one symbol in the expression,
+    # try to write it on a simplified form
+    free = list(filter(lambda x: x.is_real is not False, r.free_symbols))
+    if len(free) == 1:
+        try:
+            x = free.pop()
+            dif = r.lhs - r.rhs
+            m, b = linear_coeffs(dif, x)
+            if m.is_zero is False:
+                if m.is_negative:
+                    # Dividing with a negative number, so change order of arguments
+                    # canonical will put the symbol back on the lhs later
+                    r = r.function(-b / m, x)
+                else:
+                    r = r.function(x, -b / m)
+            else:
+                r = r.function(b, S.Zero)
+        except ValueError:
```

Comment:
We should try to keep try/except blocks as narrow as possible so we don't catch unintended exceptions. This is particularly important when catching a completely generic exception like `ValueError` that can come from anywhere.

Ideally we should use explicit checks for whatever case could lead to an error rather than catching the error like this.

Reply:
I do wish to improve this, but I just borrowed this chunk from `core/relational` so I don't know how to fix it. I would appreciate any help!